docs/learn/advanced/00-baseapp.md

```diff
@@ -408,6 +418,10 @@ Click [here](../beginner/04-gas-fees.md#antehandler) for more on the `anteHandle
 
 `RunMsgs` is called from `RunTx` with `runTxModeCheck` as parameter to check the existence of a route for each message the transaction, and with `execModeFinalize` to actually process the `sdk.Msg`s.
 
+After `CheckTx` exits, `FinalizeBlock` continues to run [`runMsgs`](../advanced/00-baseapp.md#runtx-antehandler-runmsgs-posthandler) to fully execute each `Msg` within the transaction. Since the transaction may have messages from different modules, `BaseApp` needs to know which module to find the appropriate handler. This is achieved using `BaseApp`'s `MsgServiceRouter` so that it can be processed by the module's Protobuf [`Msg` service](../../build/building-modules/03-msg-services.md). 
```

Comment:
This is not accurate, CheckTx is ran when including the tx to the mempool, not before FinalizeBlock. These 2 are in different flows, maybe just mention that FinalizeBlock also calls runmsgs and remove the `After `CheckTx` exits,`

Reply:
Ugh this is actually something prexisting from the intro page and was pasted here :/ and missed this. I think a lot is outdated